test/happyapi/oauth2/auth_test.clj

```diff
@@ -31,3 +31,13 @@
           (is credentials)
           (is (middleware/success? (auth/revoke-token request config credentials)))
           (credentials/delete-credentials provider "user"))))))
+
+(deftest scopes-test
```

Comment:
I decided to test
* the happy path: 1 overlapping scope
* no overlapping scopes
* empty scopes in token
* empty list of required scopes